### PR TITLE
Increase app minimum supported macOS version from 10.11 to 10.13

### LINKED ISSUE #57805: MacOS Desktop build: Default font of material (SF UI) does not work on Mac OS 10.12
## Steps to Reproduce

Building a MacOS desktop app with default ThemeData (unchanged TargetPlatform) and running it on MacOS 10.12 seems to produce font problems.

Unfortunately I don't have a chance to test on MacOS 10.12, [but a user report a weird font problem](https://github.com/authpass/authpass/issues/74): 
82554506-c27e8200-9b6e-11ea-88a7-ff002df81a4a

I have then tried to use Google Fonts to load alternative fonts, and this solved the issue. So it's definitely a problem with the default font of the theme.

Not sure if this is something which has to be fixed in flutter, since desktop is still just a preview and not sure if MacOS 10.12 should be "supported"(?). But I don't quite understand the problem, as the default theme on TargetPlatform.macOS uses SF UI and afaik this should be available since 10.11.

### LINKED ISSUE #107688: macOS build warning on Xcode 14 beta - The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.12, but the range of supported deployment target versions is 10.13 to 13.0.99.
On Xcode 14 beta 3
```
macos/Runner.xcodeproj warning build: The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.12, but the range of supported deployment target versions is 10.13 to 13.0.99.
```

See https://github.com/flutter/flutter/issues/40109 for previous iOS version of this issue.

```
$ flutter doctor -v
[!] Flutter (Channel unknown, 3.1.0-0.0.pre.1665, on macOS 12.4 21F79 darwin-arm, locale en)
    ! Flutter version 3.1.0-0.0.pre.1665 on channel unknown at /Users/m/Projects/flutter
    ! Upstream repository unknown
    • Framework revision 87e673c431 (7 minutes ago), 2022-07-14 17:26:22 -0700
    • Engine revision 5cc73a1274
    • Dart version 2.18.0 (build 2.18.0-283.0.dev)
    • DevTools version 2.15.0


[✓] Xcode - develop for iOS and macOS (Xcode 14.0)
    • Xcode at /Users/m/Applications/Xcode-14_beta-3.app/Contents/Developer
    • Build 14A5270f
    • CocoaPods version 1.11.3
```

### PR DESCRIPTION
**Pull request by jmagman**
Increase minimum Flutter app target macOS version from 10.11 to macOS 10.13 High Sierra and higher.

1. Change templates to `MACOSX_DEPLOYMENT_TARGET` and Podfile `platform :osx` to 10.13.
1. Add migrator for these settings for existing apps.
1. Run migrator on all example apps and integration tests.

This does not include upping the plugin templates, as those need to wait until the 10.13 framework reaches stable so new plugins can actually build with stable `flutter create`d apps, see https://github.com/flutter/flutter/pull/89695 for details.

Similar iOS migration at https://github.com/flutter/flutter/pull/101963
Fixes https://github.com/flutter/flutter/issues/107688
Fixes https://github.com/flutter/flutter/issues/57805
Part of https://github.com/flutter/flutter/issues/107492. 
Will unblock https://github.com/flutter/buildroot/pull/588

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by dnfield — APPROVED** [review]

**Comment by jmagman** [comment]
@zanderso any timing concerns with this?  Or is this okay to merge?

File: packages/flutter_tools/test/general.shard/macos/macos_project_migration_test.dart [hunk 1]
```diff
@@ -13,80 +14,81 @@ import 'package:test/fake.dart';
...
-    expect(xcodeProjectInfoFile.lastModifiedSync(), projectLastModified);
-    expect(xcodeProjectInfoFile.readAsStringSync(), contents);
-
-    expect(testLogger.statusText, isEmpty);
-  });
-
-  testWithoutContext('skips migrating script with embed', () {
-    const String contents = r'''
+  group('remove link and embed migration', () {
+    late TestUsage testUsage;
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/macos/macos_project_migration_test.dart:18]: Added the `group` so this part is a whitespace only change.

File: packages/flutter_tools/test/general.shard/macos/macos_project_migration_test.dart [hunk 2]
```diff
@@ -97,64 +99,156 @@ keep this 1
...
+        const TestUsageEvent('macos-migration', 'remove-frameworks', label: 'failure'),
+      ));
+    });
   });
 
-  testWithoutContext(
-      'migration fails with leftover FlutterMacOS.framework reference', () {
-    xcodeProjectInfoFile.writeAsStringSync('''
-				33D1A10522148B93bogus /* FlutterMacOS.framework in Bundle Framework */,
+  group('update deployment target version', () {
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/macos/macos_project_migration_test.dart:158]: These are the new tests.

**Review by zanderso — APPROVED** [review]
No timing concerns.

**Comment by jmagman** [comment]
Google testing is stalled due to omg/55671, I was told to merge manually 🙂

**Review by cbracken — COMMENTED** [review]
Thanks!

**Pull request merged**